experimental/virtio/src/vsock/socket.rs

```diff
@@ -344,6 +356,7 @@ impl ciborium_io::Write for Socket {
 
     fn flush(&mut self) -> Result<(), Self::Error> {
         // We always flush on write, so do nothing.
+        // TODO(#2876): Don't always flush on write.
```

Comment:
pre-existing: This is a bit confusing. Why do we have the flush method in the first place then, if it does nothing? 

And, just because: would it hurt to actually flush? I mean, we promise to flush...

Reply:
`flush` is a required function on the `Read` trait. The current implementation flushes every time we write some bytes, which is not always needed. This means that we flush twice for a single frame (once for the header, once for the bofy) The better implementation would use a buffer to only flush once the frame header and body has been written.

Updated the comment to try to clarify.